gslib/utils/ls_helper.py

```diff
@@ -371,11 +371,9 @@ def ExpandUrlAndPrint(self, url):
       # User provided a prefix or object URL, but it's impossible to tell
       # which until we do a listing and see what matches.
       top_level_iterator = PluralityCheckableIterator(
-          self._iterator_func(
-              url.CreatePrefixUrl(wildcard_suffix=None),
-              all_versions=self.all_versions).IterAll(
-                  expand_top_level_buckets=True,
-                  bucket_listing_fields=self.bucket_listing_fields))
+          self._iterator_func(str(url), all_versions=self.all_versions).IterAll(
```

Comment:
I think str(url) is causing issues here. Try replacing with `url.url_string`.

Reply:
Done.